sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/AttributesMap.java

```diff
@@ -17,84 +17,83 @@
 package io.opentelemetry.sdk.trace;
 
 import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.Attributes;
 import io.opentelemetry.common.ReadableAttributes;
 import java.util.HashMap;
 import java.util.Map;
 import javax.annotation.Nullable;
 
 /**
- * A map implementation with a fixed capacity that drops attributes when the map gets full.
+ * A map with a fixed capacity that drops attributes when the map gets full.
  *
- * <p>Some APIs may have slightly different behaviors, like `put` which returns null if out of
- * capacity.
+ * <p>Note: this doesn't implement the Map interface, but behaves very similarly to one.
  */
-final class AttributesMap extends HashMap<String, AttributeValue> implements ReadableAttributes {
+final class AttributesMap implements ReadableAttributes {
+  private final Map<String, AttributeValue> data = new HashMap<>();
 
   private final long capacity;
   private int totalAddedValues = 0;
-  // Here because -Werror complains about this: [serial] serializable class AttributesWithCapacity
-  // has no definition of serialVersionUID. This class shouldn't be serialized.
-  private static final long serialVersionUID = 42L;
 
   AttributesMap(long capacity) {
     this.capacity = capacity;
   }
 
-  @Nullable
-  @Override
-  public AttributeValue put(String key, AttributeValue value) {
+  public void put(String key, AttributeValue value) {
```

Comment:
I think it's worth returning the existing item like `Map` since it's familiar and opens more use cases without extra work. It's not backwards-compatible to change return value of a method unfortunately.

Reply:
That's fair. I'll make it so.

Reply:
although, this is an internal, non-public class, so I don't think it actually matters much. :)

Reply:
Oops haha. Yeah then no worries at all